services/auth/source/ldap/source_search.go

```diff
@@ -111,28 +112,69 @@ func (source *Source) findUserDN(l *ldap.Conn, name string) (string, bool) {
 func dial(source *Source) (*ldap.Conn, error) {
 	log.Trace("Dialing LDAP with security protocol (%v) without verifying: %v", source.SecurityProtocol, source.SkipVerify)
 
-	tlsConfig := &tls.Config{
-		ServerName:         source.Host,
-		InsecureSkipVerify: source.SkipVerify,
-	}
+	ldap.DefaultTimeout = time.Second * 10
+	// Remove any extra spaces in HostList string
+	tempHostList := strings.ReplaceAll(source.HostList, " ", "")
+	// HostList is a list of hosts separated by commas
+	hostList := strings.Split(tempHostList, ",")
 
-	if source.SecurityProtocol == SecurityProtocolLDAPS {
-		return ldap.DialTLS("tcp", net.JoinHostPort(source.Host, strconv.Itoa(source.Port)), tlsConfig)
+	type result struct {
+		conn *ldap.Conn
+		err  error
 	}
 
-	conn, err := ldap.Dial("tcp", net.JoinHostPort(source.Host, strconv.Itoa(source.Port)))
-	if err != nil {
-		return nil, fmt.Errorf("error during Dial: %w", err)
-	}
+	results := make(chan result, len(hostList))
+
+	for _, host := range hostList {
+		go func(host string) {
```

Comment:
I think it's better to use a standalone function rather than a closure function.

Reply:
agreed, refactored.